src/lsm/table.zig

```diff
@@ -677,13 +677,16 @@ pub fn TableType(
             if (result.exact) {
                 const value = &values[result.index];
                 if (constants.verify) {
+                    for (values[0..result.index]) |*value_before| {
+                        assert(compare_keys(key, key_from_value(value_before)) != .eq);
+                    }
```

Comment:
I'm going to pull this out for now and see if I can add this check as a separate fuzzer. That way it doesn't slow all the other tests down.

Reply:
> @batiati, I _think_ you hit the "O(N) binary search if `constants.verify`" problem. Do you remeber which "benchmark" shows the slowdown, so that we can check it here?

Sorry, I missed this notification!

It was the fuzz_tree and fuzz_forest tests where I noticed a huge slowdown caused by O(N) in binary search when `constants.verify == true`.
For reference https://github.com/tigerbeetle/tigerbeetle/pull/989#discussion_r1266670365